src/lambda/wizards/samDeployWizard.ts

```diff
@@ -357,23 +388,82 @@ export class DefaultSamDeployWizardContext implements SamDeployWizardContext {
                     'AWS.samcli.deploy.s3Bucket.prompt',
                     'Enter the AWS S3 bucket to which your code should be deployed'
                 ),
+                value: initialValue || '',
+                matchOnDetail: true,
                 ignoreFocusOut: true,
-                prompt: localize(
-                    'AWS.samcli.deploy.s3Bucket.region',
-                    'S3 bucket must be in selected region: {0}',
+            },
+            // preloaded "busy screen" message
+            // using a const in case we want to make this clickable as a way to abort the lookup.
+            items: [
+                {
+                    label: loadingBuckets
+                }
+            ],
+        })
+
+        // throw up a busy screen while the S3 buckets are loading
+        picker.makeQuickPickBusy(quickPick)
+        quickPick.show()
+
+        let listBucketsResponse: S3.ListBucketsOutput | undefined
+        let bucketMessage: string | undefined
+        let bucketError: string | undefined
+        // does the cache have any entries? If so, don't list buckets
+        if (!this.populatedBucketsByRegion) {
+            try {
+                listBucketsResponse = await s3Client.listBuckets()
+
+                // sort through buckets and add to cache if you had to run a listBuckets
+                if (listBucketsResponse && listBucketsResponse.Buckets) {
+                    // listBuckets lists all buckets tied to an account regardless of region
+                    // we need to filter the list ourselves since we do care about the region
+                    await addBucketsToRegionMap(listBucketsResponse.Buckets, this.bucketsByRegion, s3Client)
+                    // you've successfully added buckets to regions. Cool. Now don't do it again.
+                    this.populatedBucketsByRegion = true
+                }
+
+            } catch (err) {
+                // user doesn't have access to region for some reason
+                // change bucketError message
+                const error = err as Error
+                bucketMessage = localize(
```

Comment:
Is this catch encountering errors with ANY region, not just `selectedRegion`? `addBucketsToRegionMap` is working with all buckets. Perhaps `addBucketsToRegionMap` should be catching and added an error entry to the region map.

Reply:
This catches ANY errors from S3 calls. This will only error the first time we pull buckets (since it's cached and not refreshed for the duration of the wizard). 

I can verify, but I think this fails specifically due to the region that you're making the S3 call in since we feed in the S3 region from the region selection. This means that if you have a valid region code, successfully pull buckets, and go back and change a prohibited region, you'll get the standard "no buckets found in region" message rather than this error message.